instrumentation/grpc-1.6/library/src/main/java/io/opentelemetry/instrumentation/grpc/v1_6/GrpcAttributesExtractor.java

```diff
@@ -29,5 +46,14 @@ public void onEnd(
     if (status != null) {
       attributes.put(SemanticAttributes.RPC_GRPC_STATUS_CODE, status.getCode().value());
     }
+
+    if (requestMetadataValuesToCapture != null) {
+      for (String key : requestMetadataValuesToCapture) {
+        internalSet(
```

Comment:
`internalSet()` is an internal method - please try not to use it outside of the `instrumentation-api`/`instrumentation-api-semconv` modules

Reply:
True, replacing it.